### ISSUE TITLE
CodeXL - is an invalid application (OpenCL tracing)

### ISSUE THREAD

**Issue by rNoz** [GPUOpen-Tools/CodeXL#177]
Hi,

I write here because I cannot ask in https://community.amd.com/community/devgurus/codexl, and I don't know why.

I choose for example the /opt/AMDAPP/SDK/samples/opencl/cl/1.x/BinomialOption.

I compile it, and I can run it (at least with the cpu: ./BinomialOption --device cpu).

```
ldd bin/x86_64/Release/BinomialOption
        linux-vdso.so.1 (0x00007fff831ae000)
        libOpenCL.so.1 => /usr/lib/libOpenCL.so.1 (0x00007fb7b4cf3000)
        librt.so.1 => /usr/lib/librt.so.1 (0x00007fb7b4aeb000)
        libstdc++.so.6 => /usr/lib/libstdc++.so.6 (0x00007fb7b4763000)
        libm.so.6 => /usr/lib/libm.so.6 (0x00007fb7b4413000)
        libgcc_s.so.1 => /usr/lib/libgcc_s.so.1 (0x00007fb7b41fb000)
        libc.so.6 => /usr/lib/libc.so.6 (0x00007fb7b3e43000)
        libdl.so.2 => /usr/lib/libdl.so.2 (0x00007fb7b3c3b000)
        /lib64/ld-linux-x86-64.so.2 => /usr/lib64/ld-linux-x86-64.so.2 (0x00007fb7b5133000)
        libpthread.so.0 => /usr/lib/libpthread.so.0 (0x00007fb7b3a1b000)

file bin/x86_64/Release/BinomialOption
bin/x86_64/Release/BinomialOption: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 3.2.0, BuildID[sha1]=7ae316748e2d45239a4491863a3b791115f8f4b9, with debug_info, not stripped

```

Then, in CodeXL I select this project and try to use "Application Timeline Trace" because I am interested in the OpenCL calls. I set the arguments `--device cpu` and run `GPU Profile`.

It fails:

```
Unable to gather profile data. This error can occur for one of several reasons:
The active project is not an OpenCL program.
The active project does not compile or run properly (try running it manually).
You do not have write access to the profile output directory.
The application did not call any of the APIs enabled in the "API to trace" option on the "Application Timeline Trace" project setting page
The application terminated unexpectedly before the profiler wrote out any data. Try reducing the "Interval at which to write trace data during program execution" on the "Application Timeline Trace" project setting page.
```

And in the shell it says: `BinomialOption is not a valid application`.

I tried changing the Profiling options with API to trace, changing interval to 10ms, etc.. but with the same error.

If I select CPU: Time-based sampling it works, but it does not shown anything related with OpenCL Traves/calls.

When doing the CPU time based sampling the program is executed:

```
 CodeXL
/opt/AMDAPP/CodeXL/CodeXL
Platform 0 : Advanced Micro Devices, Inc.
Platform 1 : Mesa
Platform 2 : Intel(R) Corporation
Platform found : Advanced Micro Devices, Inc.

Selected Platform Vendor : Advanced Micro Devices, Inc.
Device 0 : AMD FX-8320E Eight-Core Processor Device ID is 0x5561a2011420
Executing kernel for 1 iterations
-------------------------------------------

Output
6.57815 0.920039 5.39851 5.70162 8.26097 0.291796 0.649763 5.12428 0.45692 2.12934 1.46176 2.98246 0.77529 1.75483 9.32185 8.37519 3.0711 4.25723 0.258444 2.70962 2.92678 0.370864 0.260227 5.81746 0.259135 0.955721 0.264069 0.296319 10.6425 0.323491 1.75076 6.55468 2.77851 0.511058 3.09514 1.85072 1.58886 9.89036 0.500164 5.18077 1.87278 5.15439 0.951755 7.76295 0.472647 0.720721 5.88988 8.44764 0.549665 9.24362 1.86602 0.391895 0.284768 3.444 7.73203 0.70554 0.635636 2.6687 1.31662 0.655975 0.3612 9.82934 8.02458 6.81611 0.427025 1.99273 0.824033 4.98157 1.7472 3.50752 1.91706 1.44903 1.17936 8.77909 8.75176 4.31648 0.475526 4.60656 3.12702 0.727503 3.80143 0.263592 1.19569 7.48829 6.34012 0.630731 0.342777 7.80752 0.711698 3.78331 9.43797 2.49388 3.3618 6.99124 1.19213 8.57418 0.941686 6.03544 3.74705 8.24342 1.50097 0.319403 9.26846 8.47584 0.256544 7.51195 3.14171 1.14252 2.86502 0.466197 5.45498 0.548358 1.24219 0.337471 0.279814 0.452629 2.15375 1.04632 0.26571 8.13918 0.312184 0.266537 1.60419 4.98577 10.2299 8.86188 3.75053 0.863039 4.79853 0.793155 0.505213 0.349136 2.44674 0.374217 0.257545 4.49975 0.26739 5.60221 0.26256 4.71755 0.491015 9.26468 0.910557 1.82623 0.270249 0.364942 5.68861 4.50818 3.3512 9.73913 3.12041 4.97279 0.349989 0.26132 1.81433 0.452308 0.54766 0.301856 1.34315 6.13789 2.32536 4.89929 0.928273 0.259558 10.6738 0.301363 7.72478 0.267396 10.6092 0.864408 7.26471 0.515425 3.77488 8.55132 2.55454 0.273482 0.262046 0.906033 8.2956 6.13828 0.749702 2.11485 2.38974 1.28054 3.79428 0.323309 1.90975 4.92942 0.537757 10.4465 2.36351 7.43017 4.76444 2.98303 1.66078 4.76425 6.42707 8.61073 7.32806 6.37943 10.0786 4.69597 8.05342 10.1973 3.49579 1.6189 0.262507 6.35762 7.70072 0.46485 3.25764 0.382169 2.99021 0.343108 3.99519 0.580928 0.624746 0.347488 0.494905 3.03674 0.332952 3.2767 1.73135 9.85297 0.463261 2.05304 4.28888 0.286053 1.41703 2.53896 9.10884 1.2689 0.654136 6.74399 1.15896 3.84023 0.688985 2.60849 6.42721 0.352316 3.61914 1.50455 1.49675 0.539964 4.17441 0.275324 2.89275 1.36914 1.03121 3.92367 3.59653 3.09613 0.698037 0.277152 2.73547 2.96059
```

What am I doing wrong?

**Comment by chesik-amd** [comment]
Which version of CodeXL are you using?  There was a similar issue resolved in RCP 5.2 (which is included in CodeXL 2.5).  If you are not using CodeXL 2.5, can you try out 2.5 and see if this issue still occurs?

The original discussion on this issue can be seen here:  https://github.com/GPUOpen-Tools/RCP/issues/2

**Comment by rNoz** [comment]
Thank you very much. Now it works with the AMD CPU. I needed to install the 2.5.
